src/wasm/cimbar_js/wasm.cpp

```diff
@@ -39,34 +41,43 @@ int initialize_GL(int width, int height)
 	return 1;
 }
 
+// render() and next_frame() could be put in the same function,
+// but it seems cleaner to split them.
+// in any case, we're concerned with frame pacing (some encodes take longer than others)
 int render()
 {
 	if (!_window or !_fes)
 		return 0;
 
-	// we generate 2x the amount of required blocks -- unless everything fits in a single frame.
+	if (_next)
+	{
+		_window->show(*_next, 0);
+		_window->shake();
+		return 1;
+	}
+	return 0;
+}
+
+int next_frame()
+{
+	if (!_window or !_fes)
+		return 0;
+
+	// we generate 8x the amount of required blocks -- unless everything fits in a single frame.
 	unsigned required = _fes->blocks_required();
 	if (required > cimbar::Config::fountain_chunks_per_frame())
-		required = required*4;
+		required = required*8;
```

Comment:
Decided to increase the amount of fountain chunks before we loop around. There's a somewhat squishy point at which weird things happen (large file -> 8x blocks -> exceed wirehair's uint16_t limit for block count -> disaster???) that needs further investigation, and probably a sanity check.